devel/libsdl/files/blank-window.patch

```diff
@@ -0,0 +1,130 @@
+Fix blank/black/gray window issues on Mojave and later.
+
+https://bugzilla.libsdl.org/show_bug.cgi?id=4788
+https://hg.libsdl.org/SDL/rev/ab7529cb9558
+--- src/video/quartz/SDL_QuartzVideo.m.orig	Sun Mar 22 22:03:38 2020 -0400
++++ src/video/quartz/SDL_QuartzVideo.m	Sun Apr 05 08:54:07 2020 -0700
+@@ -940,6 +940,10 @@
+     /* Set app state, hide cursor if necessary, ... */
+     QZ_DoActivate(this);
+ 
++    [ window_view setNeedsDisplay:YES ];
++  	[ [ qz_window contentView ] setNeedsDisplay:YES ];
++  	[ qz_window displayIfNeeded ];
++    
+     return current;
+ 
+     /* Since the blanking window covers *all* windows (even force quit) correct recovery is crucial */
+@@ -1115,7 +1119,11 @@
+ 
+     /* Save flags to ensure correct teardown */
+     mode_flags = current->flags;
+-
++    
++    [ window_view setNeedsDisplay:YES ];
++   	[ [ qz_window contentView ] setNeedsDisplay:YES ];
++   	[ qz_window displayIfNeeded ];
++    
+     /* Fade in again (asynchronously) if we came from a fullscreen mode and faded to black */
+     if (fade_token != kCGDisplayFadeReservationInvalidToken) {
+         CGDisplayFade (fade_token, 0.5, kCGDisplayBlendSolidColor, kCGDisplayBlendNormal, 0.0, 0.0, 0.0, FALSE);
+@@ -1155,7 +1163,13 @@
+ 
+     if (qz_window != nil) {
+         nsgfx_context = [NSGraphicsContext graphicsContextWithWindow:qz_window];
+-        [NSGraphicsContext setCurrentContext:nsgfx_context];
++        if (nsgfx_context != NULL) {
++        	[NSGraphicsContext setCurrentContext:nsgfx_context];
++      	}
++      	else {
++      		/* Whoops, looks like Mojave doesn't support this anymore */
++      		fprintf(stderr,"Unable to obtain graphics context for NSWindow (Mojave behavior)\n");
++      	}
+     }
+ 
+     /* Setup the new pixel format */
+@@ -1500,10 +1514,17 @@
+     }
+ }
+ 
+-static void QZ_UpdateRects (_THIS, int numRects, SDL_Rect *rects)
+-{
++static SDL_VideoDevice *last_this = NULL;
++
++void QZ_UpdateRectsOnDrawRect(/*TODO: NSRect from drawRect*/) {
++	// HACK
++	SDL_VideoDevice *this = last_this;
++
++	if (this == NULL) return;
++  if (SDL_VideoSurface == NULL) return;
++
+     if (SDL_VideoSurface->flags & SDL_OPENGLBLIT) {
+-        QZ_GL_SwapBuffers (this);
++// TODO
```

Comment:
Doesn't exactly inspire confidence.

Reply:
> Doesn't exactly inspire confidence.

While I agree, this is what they committed. I've asked them in the bug report if they can improve it further to resolve the "TODO"s and "HACK"s.